PWGDQ/Core/VarManager.h

```diff
@@ -2752,10 +2818,118 @@ void VarManager::FillTrackMC(const U& mcStack, T const& track, float* values)
       values[kMCMotherPdgCode] = mother.pdgCode();
     }
   }
-
   FillTrackDerived(values);
 }
 
+template <typename T, typename T1>
+void VarManager::FillEnergyCorrelators(T const& track, T1 const& t1, float* values)
+{
+  // energy correlators
+  float p2 = t1.p();
+  float E_hadron = t1.e();
+  float MassPion = o2::constants::physics::MassPionCharged;
+  float MassHadron = TMath::Sqrt(E_hadron * E_hadron - p2 * p2);
+  ROOT::Math::PtEtaPhiMVector v1(track.pt(), track.eta(), track.phi(), o2::constants::physics::MassJPsi);
```

Comment:
The construction of this vector is hardcoded always for jpsi. I think that this should be optional. You can use the candidate type parameter from the process function to derive which mass assumption to do here.